src/data-loading.test.tsx

```diff
@@ -9,44 +9,24 @@ import { mount, configure } from 'enzyme'
 
 configure({ adapter: new Adapter() })
 
-const lengthCalculatorDataDefinition: DataDefinition<{ dataArg: string }, number, {}> = {
-    loadData: props =>
-        new Promise(resolve =>
-            setTimeout(() => {
-                resolve(props.dataArg.length)
-            }),
-        ),
-}
-
-// Test component with data
-export const TestComponentWithData: React.FC<{ length: number | undefined }> = ({ length }) => (
-    <div>{length ? `Length: ${length}` : 'Loading'}</div>
-)
-
-const resources = new DataLoaderResources<{}>()
-const { createRegisterableComposition } = getRegistrationCreators<{}>()
-const { middleware, createRegisterableComponentWithData } = init<{}>(resources)
-
-export const testCompositionRegistration = createRegisterableComposition<'main'>()(
-    'test-composition',
-    contentAreas => <TestComposition main={contentAreas.main} />,
-)
+it('can load data for component', async () => {
+    const resources = new DataLoaderResources<{}>()
+    const { middleware, createRegisterableComponentWithData } = init<{}>(resources)
```

Comment:
how about `createDataloaderLayout`?

Reply:
Will do in another PR as that is a breaking change

Reply:
Create a new issue with suggestions :)